### PR TITLE
Insert a newline if the first element is multiline

### PR DESCRIPTION
**Pull request by deecewan**
There is an inconsistency between certain cops with respect to adding in a newline.

For instance, the code:

```ruby
[
  {
    a: 1,
    b: 2,
  }]
```

when run with `Layout/MultilineArrayBraceLayout` set to `new_line`, will produce

```ruby
[
  {
    a: 1,
    b: 2,
  }
]
```

However, 

```ruby

[{
  a: 1,
  b: 2,
  }
]
```

produces no errors.

I believe that this behavior being inconsistent makes it difficult to correctly configure a style.

I initially came across this behavior due to wanting

```ruby
[{
  a: 1,
  b: 2,
}]
```
but the cop's I was using were changing it to

```ruby
[{
  a: 1,
  b: 2,
},]
```

If that's the intended effect, given that 'technically' the array is multiline, it would make more sense to have the multiline rules also apply to the start of the array. It seems as though there is consistent behavior and the end of arrays, but not the starts.  Otherwise, if this is not intended, the same rules should apply to the end of the array (only add the trailing comma if the closing bracket is on a different line).

In this pull request, I changed the children line-break method from checking for 2 children (and then adding the line-break) to checking for at least 1 child before adding the line break.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
@jonas054 Can you take a look at this?

**Comment by jonas054** [comment]
The code changes look good, and all old spec examples still pass.

It's a user-visible change, so we need a change log entry.

**Comment by jonas054** [comment]
Something weird happened here, git-wise. Do you see how to fix it?

**Comment by deecewan** [comment]
i have no idea what's happened here :/
any ideas?

**Comment by jonas054** [comment]
You can resolve it by doing
```sh
git fetch upstream
git checkout -b 4574_insert_newline upstream/master
git cherry-pick "<hash of your 1st commit for this PR>"
git cherry-pick "<hash of your 2nd commit for this PR>"
git cherry-pick "<hash of your 3rd commit for this PR>"
git push origin 4574_insert_newline upstream
```
I think a new PR will be created, but that's ok. This way is easier than trying to fix the problems on `master`. I think it's generally better to use feature branches that to work directly on `master`.

**Comment by deecewan** [comment]
@jonas054 I went back and sorted this out with a couple of force pushes - we are back where we should be now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,6 +52,20 @@
 * [#4664](https://github.com/bbatsov/rubocop/pull/4664): Fix typos in Rails/HttpPositionalArguments. ([@JoeCohen][])
 * [#4618](https://github.com/bbatsov/rubocop/pull/4618): Fix `Lint/FormatParameterMismatch` false positive if format string includes `%%5B` (CGI encoded left bracket). ([@barthez][])
 * [#4604](https://github.com/bbatsov/rubocop/pull/4604): Fix `Style/LambdaCall` to autocorrect `obj.call` to `obj.`. ([@iGEL][])
+* [#4574](https://github.com/bbatsov/rubocop/pull/4574): Fix an inconsistency in multiline array
+layout cops. ([@deecewan][])
```
Comment (jonas054) [inline CHANGELOG.md]: There's an unwanted line break here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -52,6 +52,20 @@
...
+* New cop `ElseLayout` checks for odd arrangement of code in the `else` branch of a conditional expression. ([@bbatsov][])
```
Comment (jonas054) [inline CHANGELOG.md]: You seem to have included a bunch of log entries from other changes.

**Comment by bbatsov** [comment]
Closing due to lack of progress. I don't have time to finish this myself, but if @jonas054 @Drenmi @pocke @koic want to tackle it, that'd be great.

**Pull request closed without merging**